<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Baxter Worthing">

<title>Risk of Maladaptive Phenological Responses to Climate Change in Natural Populations of Hybrid Poplar Trees</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="Main_files/libs/clipboard/clipboard.min.js"></script>
<script src="Main_files/libs/quarto-html/quarto.js"></script>
<script src="Main_files/libs/quarto-html/popper.min.js"></script>
<script src="Main_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Main_files/libs/quarto-html/anchor.min.js"></script>
<link href="Main_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Main_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Main_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Main_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Main_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Risk of Maladaptive Phenological Responses to Climate Change in Natural Populations of Hybrid Poplar Trees</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Baxter Worthing </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>For trees inhabiting temperate regions of the world, traits related to seasonal phenology are often adapted to home environment, suggesting that rapid changes to these local climates could cause populations to become maladapted in the near future. For instance, the timing of the initiation of leaf expansion from formerly dormant buds in spring (a process here on referred to as “bud flush”) represents an adaptive balance between selection for a long growing season and selection against the risk of cold injury <span class="citation" data-cites="liu2018 henner2020popfor richards2020 gao2022">(<a href="#ref-liu2018" role="doc-biblioref">Liu et al. 2018</a>; <a href="#ref-henner2020popfor" role="doc-biblioref">Henner et al. 2020</a>; <a href="#ref-richards2020" role="doc-biblioref">Richards et al. 2020</a>; <a href="#ref-gao2022" role="doc-biblioref">Gao et al. 2022</a>)</span>. In many tree species, bud flush timing is known to be strongly influenced by changes in mean daily temperatures during late winter and early spring (<span class="citation" data-cites="horvath2003">(<a href="#ref-horvath2003" role="doc-biblioref">Horvath et al. 2003</a>)</span>;<span class="citation" data-cites="pau2011predicting">(<a href="#ref-pau2011predicting" role="doc-biblioref">Pau et al. 2011</a>)</span>;<span class="citation" data-cites="vitasse2014">(<a href="#ref-vitasse2014" role="doc-biblioref">Vitasse, Lenz, and KÃ¶rner 2014</a>)</span>;<span class="citation" data-cites="flynn2018temperature">(<a href="#ref-flynn2018temperature" role="doc-biblioref">Flynn and Wolkovich 2018</a>)</span>). This environmental sensitivity means that long-term changes or short-term fluctuations in the environmental cues that influence bud flush timing may cause trees to undergo bud flush too early or too late in the season. This mistiming of phenology could reduce fitness through exposure to late spring frost events or shortening of the realized growing season (Way, 2011; Brunner et al., 2014; Charrier et al., 2015). While some trees may be able to adjust bud flush timing to track changes in the optimal growing season, recent research has found mixed support for this hypothesis across tree taxa (Gao, Lian and Liu et al., 2022). Additionally, research in multiple tree species shows evidence of genetic variation for bud flush timing, implying that both genetic and environmental factors influence this trait [<span class="citation" data-cites="olson2012">Olson et al. (<a href="#ref-olson2012" role="doc-biblioref">2012</a>)</span>; Alder et al., 2021]. Accurate prediction of the risk of maladaptive phenology in future climates requires a better understanding of the relationship between genetic variation, environmental context and bud flush timing. Furthermore, predictions of maladptive risk could facilitate prioritization of tree populations for conservation concern, based on genetic background and local environment.</p>
<p>Association studies have identified many of the specific loci that influence bud flush timing, which include genes in the flowering time pathway, phytochromes, DAM genes, and genes related to signaling and response to temperature (Olson et al., 2013; Howe et al., 2015; McKown et al 2014; McKown et al., 2018). However, these studies identify different loci across environments, populations and closely-related species, implying that genotype by environment (GxE) and genotype by genotype (GxG) interactions influence the genetic architecture of this trait. The nature of these genetic interactions have significant implications for tree conservation because trees are long lived, sessile organisms. Long generation times may allow anthropogenic climate change to progress faster than trees can evolve adaptively or migrate in order to maintain fitness in future environments (Savolainen et al., 2004; Aitken et al., 2008; Duputié et al., 2015). This means that phenotypic plasticity is an important route through which trees can maintain adaptive phenological timing in a changing climate (Vitasse et al., 2010; Scheiner et al., 2019; Cooper et al., 2019). GxE interaction can be thought of as genetic variation for the reaction norm of a plastic trait, and some genotypes may display bud flush reaction norms that are pre-adapted to climate change, allowing bud flush timing to track changes in the length of the optimal growing season over time. Similarly, introgression may act as a source of genetic novelty that could facilitate rapid evolution of phenology, but the extent to which GxG interactions in admixed individuals manifest as novel bud flush phenotypes remains unknown.</p>
<p>Hybridization is common in forest trees, and natural hybrid zones that span environmental gradients provide a unique opportunity to study these GxG and GxE interactions, as admixed individuals expose novel combinations of genotypes to new environments. Admixed genotypes may display divergent plasticity compared to parental species, but it is likely that this increased plasticity only increases capacity to adapt to future growing seasons in certain environments and populations. Therefore, properly assessing the adaptive potential of admixed tree genotypes requires sampling of broad climatic gradients and genetic gradients. Such landscape genetic analyses can help determine the potential for plasticity and introgression to drive or disrupt pre-adaptation to future climates across the landscape.</p>
<p>Here, we took advantage of natural hybrid zones between <em>Populus balsamifera</em> and <em>Populus trichocarpa</em> to determine the genetic architecture of bud flush timing in individuals with varying degrees of admixture. <em>Populus balsamifera</em> and <em>Populus trichocarpa</em> are two closely related species of cottonwood native to northern North America. Genetic analyses suggest the two species began to diverge 1.2 million years ago, but gene flow continued between them both before and after a brief period of isolation beginning 860,000 years ago (Bolte et al., 2024). Despite continued gene flow, these species have diverged ecologically. <em>Populus balsamifera</em> is a boreal tree inhabiting mountain valleys and river drainages of continental North America as far north as the subarctic. Meanwhile, <em>Populus trichocarpa</em> is limited to the more temperate, maritime climates of the Pacific Northwest and Western Canada. These two species continue to naturally hybridize in regions where their ranges overlap, and past studies have identified hybrid zones populated by both recent and advanced-generation hybrids (Suarez-Gonzalez et al., 2016; Bolte et al., 2024).</p>
<p>In <em>Populus balsamifera</em> and <em>Populus trichocarpa</em>, two physiological processes contribute to bud flush timing: ecodormancy and endodormancy. Both of these stages of winter dormancy must be released before bud flush can take place, an adaptation that likely arose as a safeguard against premature bud flush. Endodormancy is broken by the accumulation of chilling temperatures during late winter, while ecodormancy is broken by the accumulation of heat sums in early spring (Vitasse &amp; Körner, 2014). Heritable variation for chilling/warming requirements may underlie genetic and GxE effects on bud flush (Way, 2011; Guy, 2014; Lundell et al., 2019; Thibault et al., 2020). Given the climatic differences between their native ranges, P. balsamifera and P. trichocarpa likely experience divergent selection for bud flush timing, but the genetic and phenotypic consequences of this divergent selection have not been extensively studied in the hybrids of these two species. Admixture between P. balsamifera and P. trichocarpa may underlie bud flush reaction norms that will be necessary to maintain fitness under climate change, or it may disrupt existing allele combinations that are pre-adapted to future climates. Natural hybrid zones between these species offer a unique opportunity to weigh the relative probability of these possible interactions between introgesssion, plasticity and climate change.</p>
<p>In this study, our goal was to test the potential for plasticity to maintain adaptive bud flush timing in the face of climatic change, asses the influence of genetic variation on this potential, and to better understand the effect of introgression on the distribution of this variation between species and populations. With this in mind, we sampled 575 <em>Populus balsamifera</em> and <em>Populus trichocarpa</em> trees from 6 distinct transects passing through regions of range overlap between these species, which we expected to contain hybrid zones. We sequenced the full genome of each sampled tree and propagated clones of each sample in 3 climatically-distinct common gardens over multiple years, in order to observe differences in bud flush timing. We then combined these bud flush observations and genomic data with environmental variables in order to fit genomic prediction models. We used the variance components and predictions from these models to answer the following questions:</p>
<ol type="1">
<li>To what extent is bud flush timing plastic across environments?</li>
<li>Are bud flush timing reaction norms predictable across genetic and environmental context?</li>
<li>Will predicted changes in bud flush timing reduce the fitness of natural populations under future climates?</li>
<li>How does introgression influence heritable variation for bud flush timing and risk of maladptation to future climates?</li>
</ol>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<section id="sampling" class="level3">
<h3 class="anchored" data-anchor-id="sampling">Sampling</h3>
<p>In January 2020, we collected dormant branch cuttings from 546 poplar trees along 7 distinct latitudinal transects spanning natural contact zones between Populus trichocarpa and Populus balsamifera (Figure 1a).These cuttings were transported to a greenhouse at Virginia Tech (Blacksburg, VA) for propagation in a mist house. Rooted cuttings were then propagated in 3 common gardens (Blacksburg, VA; Burlington, VT and Fargo, ND). Three replicates of each of the 546 cuttings were planted in each garden, using a randomized block design, in June 2020.</p>
</section>
<section id="whole-genome-sequencing-and-variant-calling" class="level3">
<h3 class="anchored" data-anchor-id="whole-genome-sequencing-and-variant-calling">Whole genome sequencing and variant calling</h3>
<p>Whole genome sequencing and subsequent bioinformatic analyses were performed as described in Bolte et al.&nbsp;(2024). Briefly, Genomic DNA libraries for all sampled individuals were constructed at the Duke University Center for Genomic and Computational Biology and sequenced using an Illumina NovaSeq 6000 instrument with an S4 flow cell in 2 × 150 bp format with 64 samples per lane (Illumina Inc., San Diego, USA). De-indexing, QC, trimming adapter sequences, and sequence preprocessing were completed by the sequencing facility. We used the Burrows-Wheeler Aligner (BWA) to map resulting reads to the P. trichocarpa reference genome (v4.0) and called variants in gVCF format for each individual using The Genome Analysis Toolkit (GATK) (v3.7) (<span class="citation" data-cites="li2010fast">(<a href="#ref-li2010fast" role="doc-biblioref">Li and Durbin 2010</a>)</span>; <span class="citation" data-cites="van2020genomics">Van der Auwera and O’Connor (<a href="#ref-van2020genomics" role="doc-biblioref">2020</a>)</span>). We used GATK to merge individual gVCFs and then, using <em>vcftools</em>, version 0.1.16, filtered the resulting ~82 million variant sites to remove variants with missingness &gt;10%, mapping quality &lt;40, elevated strand bias (FS &gt; 40, SOR &gt; 3), high differential mapping quality or positional bias between reference and alternate alleles ( MQRankSum &lt; −12.5; ReadPosRankSum &lt; −8) or low quality-by-depth (QD &lt; 2) (<span class="citation" data-cites="danecek2011variant">(<a href="#ref-danecek2011variant" role="doc-biblioref">Danecek et al. 2011</a>)</span>). We then further filtered these variants to retain only bi-alleleic SNPs and used <em>Plink</em> v1.9 to thin variant sites in high linkage disequilibrium using 10,000bp windows sliding in increments of 1000 bp (<span class="citation" data-cites="purcell2007plink">(<a href="#ref-purcell2007plink" role="doc-biblioref">Purcell et al. 2007</a>)</span>). After these filters, 299,335 variant sites remained. Finally, as described in Bolte et al.&nbsp;(2024), we removed 29 individuals of unknown ancestry, 8 individuals without phenotype data, and one individual due to excess missing genotypoes, leaving 509 samples for further analysis. As described in Bolte et al.&nbsp;(2024), we used this set of LD-pruned SNPs to perform ADMIXTURE analysis, using values of K from 2-6 and comparing cross validation error (k-folds = 10) between these values (<span class="citation" data-cites="alexander2009">(<a href="#ref-alexander2009" role="doc-biblioref">Alexander, Novembre, and Lange 2009</a>)</span>).</p>
</section>
<section id="phenotype-observations-and-quality-control" class="level3">
<h3 class="anchored" data-anchor-id="phenotype-observations-and-quality-control">Phenotype observations and quality control</h3>
<p>Bud flush observations were performed in 2021, 2022 and 2023 for the VA and VT gardens, and only in 2021 for the ND garden. Bud flush was monitored at least two days per week through the spring each year. We scored bud flush as the number of Julian days at which the scales of terminal buds parted and leaf tips emerged, as described in McKown et al.&nbsp;(2018). To facilitate comparison of bud flush timing across years and gardens, we converted Julian day measurements to units of cumulative growing degree days (cGDD) using the formula described in Olson et al.&nbsp;(2013). Conversions were made using custom R (version 3.6.1) scripts and daymet daily climate data for each year at each garden (<span class="citation" data-cites="r2024">(<a href="#ref-r2024" role="doc-biblioref">R Core Team 2024</a>)</span>;<span class="citation" data-cites="thornton2021">(<a href="#ref-thornton2021" role="doc-biblioref">Thornton et al. 2021</a>)</span>; <span class="citation" data-cites="thornton1997">(<a href="#ref-thornton1997" role="doc-biblioref">Thornton, Running, and White 1997</a>)</span>). We also used custom R scripts to perform quality control on bud flush observations and to fit linear models calculate genetic BLUP estimates for each genotype from raw bud flush observations, accounting for the effect of each randomized block in our planting design.</p>
</section>
<section id="environmental-data" class="level3">
<h3 class="anchored" data-anchor-id="environmental-data">Environmental data</h3>
</section>
<section id="models-of-bud-flush-timing" class="level3">
<h3 class="anchored" data-anchor-id="models-of-bud-flush-timing">Models of bud flush timing</h3>
<p>To estimate the effect of environment, genetics and their interaction on bud flush timing, we fit mixed effects linear models using the R packages lme4 and nlme (<span class="citation" data-cites="bates2015">(<a href="#ref-bates2015" role="doc-biblioref">Bates et al. 2015</a>)</span>;<span class="citation" data-cites="nlme">(<a href="#ref-nlme" role="doc-biblioref">Pinheiro, Bates 2023</a>)</span>). We used <em>Plink</em> to convert the VCF of LD-pruned SNPs to a matrix and perform principle component analysis (PCA). We also performed PCA on the set of environmental variables calculated for each year in each common garden with the <em>prcomp</em> function in R. We used</p>
<p>To test for genetic clines , we also regressed common garden observations of bud flush timing on climatic varibles describing the provenances of each observed tree. We compared linear and quadradic mixed effects models of bud flush timing, including an effect of source environment and genetic admixture</p>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="genomic-prediction" class="level3">
<h3 class="anchored" data-anchor-id="genomic-prediction">Genomic prediction</h3>
<p>Table 1 shows the variance components of the three genomic prediction models we fit in this study, as well as the prediction accuracy of these models under cross-validation.</p>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>Climate change poses a risk to all life on earth, as it threatens to disrupt any existing equilibrium between adaptive genetic variation and contemporary climates, leading to local maladaptation or even extinction of certain species or populations.&nbsp;This concern is exacerbated in long lived organisms, as capacity to evolve adaptively in response to changes in local climate is reduced by long generation time.&nbsp;Two key processes can potentially insulate populations of long-lived organisms from maladaptation are plasticity and adaptive introgression. Plasticity for adaptive traits is likely to have responded to selection in the past,and this selection may manifest as GxE when it is divergent across populations. The shape of the reaction norm under future climates and the degree of genetic control over that reaction norm define the risk of maladaptation for a plastic trait. If adaptive plasticity is rare and highly heritable, then conservation efforts should be concerned with the capacity for this adaptive variation to spread between populations. Therefore, introgression must be considered as a potential conduit of genetic rescue when GxE interactions for adaptive traits are strong and divergent.</p>
<p>In this study, we observed a strong influence of GxE on bud flsuh timing among the poplar trees we sampled.</p>
<p>Poplar trees inhabit climatically variable environments and often persist through clonal propogation over timescales that exceed the lifetime of an individual tree. With this in mind, it makes sense that these trees display a high degreee of plasticity for a phenological trait. The strong GxE interactions we observe may reflect divergent selection for phenological plasticity across poplar populations.</p>
<p>In this work, we predict maladptation of phenology from genetic data across continuous and discrete environments. This framework essentially asses polygenic risk of maladptive plasticity in future climates. In past research, genome-wide estimates of maladpative risk under climate change have generally taken the form of genomic offset estimations, in which risk of maladaptation to future climates is predicted based on existing patterns of allele turnover along contemporary environmental gradients. Genomic offset is a powerful tool for risk assessment in conservation settings, but its effectiveness is likely to vary with conservation context. For example, being a purely genetic metric, GO fails to account for phenotypic plasticity, and therefore may overestimate maladaptation that could be mitigated through plastic adjustmetns to adaptive traits.However, this issue is reduced when there is a genetic component to plasticity (ie. genotype-by-environment interaction), meaning that allelic variation underlying variation in plasticity may be encapsulated in GO calculations. Our results suggest that while bud flush timing is highly plastic across envrionments, much of the plasticity in this adaptive trait is heritable, as suggested by the strong GxE effects we observe. However, The strong effect of environment in our predictive models suggest that purely genetic methods fail to fully encapsulate variation for this trait.</p>
<p>Additionally, GO is thought to overestimate maladaptation in contexts where rapid evolution is possible. The ability of GO to predict maladaptation rests on the asssumption that the genetic variation involved in adaptation to existing climatic gradients will also play a role in adaptation to future climates. If genetic diversity is high, and life history allows for selection to act swiftly on this variation, rapid evolution to future climate may be possible and may involve different loci than those selected upon by past climates. For this reason, GO is thought to be more helpful for conservation of long lived organisms, such as forest trees. However, hybridization is common in forest trees and introgression of genetic variation between tree species may facilitate rapid evolution to changing climate. Here, we sought to asses the effect of genetic admixture on bud flush reaction norms and predict the potential for these effects to maintain or reduce fitness in future climates.</p>
<section id="figures" class="level3">
<h3 class="anchored" data-anchor-id="figures">Figures</h3>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="2">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-map" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-map-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="figs/msamp.jpg" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-map-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: A
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figs/NSFg_map.jpg" class="img-fluid figure-img"></p>
<figcaption>B</figcaption>
</figure>
</div>
</div>
</div>
</div>

</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-alexander2009" class="csl-entry" role="listitem">
Alexander, David H., John Novembre, and Kenneth Lange. 2009. <span>“Fast Model-Based Estimation of Ancestry in Unrelated Individuals.”</span> <em>Genome Research</em> 19 (9): 1655–64. <a href="https://doi.org/10.1101/gr.094052.109">https://doi.org/10.1101/gr.094052.109</a>.
</div>
<div id="ref-bates2015" class="csl-entry" role="listitem">
Bates, Douglas, Martin Mächler, Ben Bolker, and Steve Walker. 2015. <span>“Fitting Linear Mixed-Effects Models Using<span><strong>lme4</strong></span>.”</span> <em>Journal of Statistical Software</em> 67 (1). <a href="https://doi.org/10.18637/jss.v067.i01">https://doi.org/10.18637/jss.v067.i01</a>.
</div>
<div id="ref-danecek2011variant" class="csl-entry" role="listitem">
Danecek, Petr, Adam Auton, Goncalo Abecasis, Cornelis A Albers, Eric Banks, Mark A DePristo, Robert E Handsaker, et al. 2011. <span>“The Variant Call Format and VCFtools.”</span> <em>Bioinformatics</em> 27 (15): 2156–58.
</div>
<div id="ref-flynn2018temperature" class="csl-entry" role="listitem">
Flynn, DFB, and EM Wolkovich. 2018. <span>“Temperature and Photoperiod Drive Spring Phenology Across All Species in a Temperate Forest Community.”</span> <em>New Phytologist</em> 219 (4): 1353–62.
</div>
<div id="ref-gao2022" class="csl-entry" role="listitem">
Gao, Shan, Eryuan Liang, Ruishun Liu, Flurin Babst, J. Julio Camarero, Yongshuo H. Fu, Shilong Piao, et al. 2022. <span>“An Earlier Start of the Thermal Growing Season Enhances Tree Growth in Cold Humid Areas but Not in Dry Areas.”</span> <em>Nature Ecology &amp; Evolution</em> 6 (4): 397–404. <a href="https://doi.org/10.1038/s41559-022-01668-4">https://doi.org/10.1038/s41559-022-01668-4</a>.
</div>
<div id="ref-henner2020popfor" class="csl-entry" role="listitem">
Henner, Dagmar N, Astley Hastings, Mark Pogson, Niall P McNamara, Christian A Davies, and Pete Smith. 2020. <span>“PopFor: A New Model for Estimating Poplar Yields.”</span> <em>Biomass and Bioenergy</em> 134: 105470.
</div>
<div id="ref-horvath2003" class="csl-entry" role="listitem">
Horvath, David P., James V. Anderson, Wun S. Chao, and Michael E. Foley. 2003. <span>“Knowing When to Grow: Signals Regulating Bud Dormancy.”</span> <em>Trends in Plant Science</em> 8 (11): 534–40. <a href="https://doi.org/10.1016/j.tplants.2003.09.013">https://doi.org/10.1016/j.tplants.2003.09.013</a>.
</div>
<div id="ref-li2010fast" class="csl-entry" role="listitem">
Li, Heng, and Richard Durbin. 2010. <span>“Fast and Accurate Long-Read Alignment with Burrows–Wheeler Transform.”</span> <em>Bioinformatics</em> 26 (5): 589–95.
</div>
<div id="ref-liu2018" class="csl-entry" role="listitem">
Liu, Qiang, Shilong Piao, Ivan A. Janssens, Yongshuo Fu, Shushi Peng, Xu Lian, Philippe Ciais, Ranga B. Myneni, Josep Peñuelas, and Tao Wang. 2018. <span>“Extension of the Growing Season Increases Vegetation Exposure to Frost.”</span> <em>Nature Communications</em> 9 (1). <a href="https://doi.org/10.1038/s41467-017-02690-y">https://doi.org/10.1038/s41467-017-02690-y</a>.
</div>
<div id="ref-olson2012" class="csl-entry" role="listitem">
Olson, Matthew S., Nicholas Levsen, Raju Y. Soolanayakanahally, Robert D. Guy, William R. Schroeder, Stephen R. Keller, and Peter Tiffin. 2012. <span>“The Adaptive Potential of <span><em>Populus Balsamifera</em> </span>L. To Phenology Requirements in a Warmer Global Climate.”</span> <em>Molecular Ecology</em> 22 (5): 1214–30. <a href="https://doi.org/10.1111/mec.12067">https://doi.org/10.1111/mec.12067</a>.
</div>
<div id="ref-pau2011predicting" class="csl-entry" role="listitem">
Pau, Stephanie, Elizabeth M Wolkovich, Benjamin I Cook, T Jonathan Davies, Nathan JB Kraft, Kjell Bolmgren, Julio L Betancourt, and Elsa E Cleland. 2011. <span>“Predicting Phenology by Integrating Ecology, Evolution and Climate Science.”</span> <em>Global Change Biology</em> 17 (12): 3633–43.
</div>
<div id="ref-nlme" class="csl-entry" role="listitem">
Pinheiro, José, Douglas Bates, and R Core Team. 2023. <span>“Nlme: Linear and Nonlinear Mixed Effects Models.”</span> <a href="https://CRAN.R-project.org/package=nlme">https://CRAN.R-project.org/package=nlme</a>.
</div>
<div id="ref-purcell2007plink" class="csl-entry" role="listitem">
Purcell, Shaun, Benjamin Neale, Kathe Todd-Brown, Lori Thomas, Manuel AR Ferreira, David Bender, Julian Maller, et al. 2007. <span>“PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses.”</span> <em>The American Journal of Human Genetics</em> 81 (3): 559–75.
</div>
<div id="ref-r2024" class="csl-entry" role="listitem">
R Core Team. 2024. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-richards2020" class="csl-entry" role="listitem">
Richards, Thomas J., Almir Karacic, Rami-Petteri Apuli, Martin Weih, Pär K. Ingvarsson, and Ann Christin Rönnberg-Wästljung. 2020. <span>“Quantitative Genetic Architecture of Adaptive Phenology Traits in the Deciduous Tree, Populus Trichocarpa (Torr. And Gray).”</span> <em>Heredity</em> 125 (6): 449–58. <a href="https://doi.org/10.1038/s41437-020-00363-z">https://doi.org/10.1038/s41437-020-00363-z</a>.
</div>
<div id="ref-thornton1997" class="csl-entry" role="listitem">
Thornton, Peter E., Steven W. Running, and Michael A. White. 1997. <span>“Generating Surfaces of Daily Meteorological Variables over Large Regions of Complex Terrain.”</span> <em>Journal of Hydrology</em> 190 (3-4): 214–51. <a href="https://doi.org/10.1016/s0022-1694(96)03128-9">https://doi.org/10.1016/s0022-1694(96)03128-9</a>.
</div>
<div id="ref-thornton2021" class="csl-entry" role="listitem">
Thornton, Peter E., Rupesh Shrestha, Michele Thornton, Shih-Chieh Kao, Yaxing Wei, and Bruce E. Wilson. 2021. <span>“Gridded Daily Weather Data for North America with Comprehensive Uncertainty Quantification.”</span> <em>Scientific Data</em> 8 (1). <a href="https://doi.org/10.1038/s41597-021-00973-0">https://doi.org/10.1038/s41597-021-00973-0</a>.
</div>
<div id="ref-van2020genomics" class="csl-entry" role="listitem">
Van der Auwera, Geraldine A, and Brian D O’Connor. 2020. <em>Genomics in the Cloud: Using Docker, GATK, and WDL in Terra</em>. O’Reilly Media.
</div>
<div id="ref-vitasse2014" class="csl-entry" role="listitem">
Vitasse, Yann, Armando Lenz, and Christian KÃ¶rner. 2014. <span>“The Interaction Between Freezing Tolerance and Phenology in Temperate Deciduous Trees.”</span> <em>Frontiers in Plant Science</em> 5 (October). <a href="https://doi.org/10.3389/fpls.2014.00541">https://doi.org/10.3389/fpls.2014.00541</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>