---
title: "Untitled"
author: "Baxter Worthing"
date: "4/11/2024"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



```{r}
source("~/Documents/plot_theme.R") # for theme_Publication

root <- "/Volumes/kellrlab/Baxter/GenomicPrediction/GP_manuscript/"
```



Need:
snp PCA
sampling map code 
var comp table 




# samples map
for some reason this code doesn't run in R version 4.4.1
```{r}
library(tidyverse)
library(rnaturalearth)
library(rnaturalearthdata)
library(sf)


sampDat <- readxl::read_xlsx("/Volumes/pophyb/Combined_Transect_Sampling_Data_2020.xlsx", na = "NA", sheet = 2) %>%
  mutate(Long = as.numeric(Long))

mapSPs  <- ne_states(c("united states of america", "canada"))

sf_use_s2(use_s2 = FALSE)
mapSPs_subS <- st_crop(mapSPs,  raster::extent(-155,-100,40,70))


ras <- raster::raster("/Volumes/kellrlab/Baxter/popClim/GRAY_50M_SR_W/GRAY_50M_SR_W.tif")

sf_use_s2(use_s2 = FALSE)
ras_subS <- raster::crop(ras,  raster::extent(-155,-100,40,70))

ras_sub_spdfS <- as(ras_subS, "SpatialPixelsDataFrame")

ras_sub_dfS <- as.data.frame(ras_sub_spdfS) %>%
  dplyr::rename("value"=1, "x"=2, "y"=3)



# need admixture data joined to sampling locations 
ids <-  read_table("/Volumes/pophyb/admixture/ids.txt",col_names = F)
# some admixture data for 900s has letter IDs
fix_ids <- read_csv("/Volumes/pophyb/numLetterIDkey.csv") %>%
  distinct() %>%
  mutate(NUM_ID=as.character(NUM_ID))
  
k2 <- read_table("/Volumes/pophyb/admixture/plink.2.Q", col_names = F)
k4 <- read_table("/Volumes/pophyb/admixture/plink.4.Q", col_names = F)

kDat <- tibble(PLANT_ID=ids$X1, kb=k2$X1, kt=k2$X2, k4a=k4$X1, k4b=k4$X2, k4c=k4$X3, k4d=k4$X4) %>%
  mutate(PLANT_ID=str_replace(PLANT_ID,"_.*","")) %>%
  dplyr::rename(ID=PLANT_ID)


dat <- left_join(sampDat, kDat, by="ID") %>%
  filter(! is.na(kt))   #!!!! 91 out of full 667 missing from kDat, leaving 575


msamp  <- ggplot(ras_sub_dfS, aes(x=x, y=y, fill=value), alpha=0.99) +
  geom_tile() +
   geom_sf(data = mapSPs_subS,aes(x=longitude, y=latitude), fill="white", color="white", linewidth=.7, alpha=0.0001) +
  #geom_polygon(data = balsShp_subS,aes(x=long, y=lat, group=group, color="P. balsamifera"), fill="#2e8de6",  color="#2e8de6", size=.25, alpha=.1) +
   #geom_polygon(data = trichShp_subS,aes(x=long, y=lat, group=group, color="P. trichocarpa"), fill="#ed411f", color="#ed411f", size=.25, alpha=.1) +
  #geom_label( data= labT, aes(label=lab, y=y, x=x),, color="#ed411f", size=16) +
   #geom_label(data= labB, aes(label=lab, y=y, x=x), color="#2e8de6", size=16) +
   geom_point(data = dat, aes(x=Long, y=Lat, group=Transect, fill=200, color=kt), size= 3.5) +
  scale_fill_gradient(high="black", low="grey", guide = "none") +
  #scale_color_gradient(low="#2e8de6", high="#ed411f") +
  scale_color_gradient2(low="#2e8de6", high="#f59e1b", mid ="#17eb57" , midpoint = 0.5) +
  labs(color="% P. trichocarpa ancestry", x="Longitude", y="Latitude") +
  theme(legend.text = element_text(size=29)) +
  theme(legend.title = element_text(size=29)) +
  theme(axis.title = element_text(size=20)) +
  theme(axis.text = element_text(size=16)) +
  theme_Publication() +
  theme(panel.grid.major = element_blank()) +
  theme(legend.key.size= unit(2, "cm")) +


ggsave(paste0(root,"figs/msamp.jpg"),msamp, device = "jpg", width=26, height = 20)

```




# common garden map

```{r}


mapSP <- ne_countries(country = c("canada","united states of america"), scale= "small")


sf_use_s2(use_s2 = FALSE)
mapSPs_sub <- st_crop(mapSPs,  raster::extent(-110,-50,34,72))

mapTidy <-mapSPs_sub 


gardenCoords <- tibble(Garden=c("Burlington, VT",  "Blacksburg, VA", "Fargo, ND"), lat=c(44.443,37.23, 46.88), long=c(-73.189, -80.41, -96.79), Years=c("2021-2023","2021-2023","2021"))

gardenMap2 <- ggplot(mapTidy, aes(x=longitude, y=latitude)) +
  geom_sf(alpha= .85, fill= "#e3caa6", color="white",linewidth=1.5) +
  geom_point(data = gardenCoords, aes(x=long, y=lat), color="red", inherit.aes = F, size=6) +
  geom_label(data = gardenCoords, aes(label= Garden, x=long, y=lat +1), size= 9, inherit.aes = F) +
   geom_text(data = gardenCoords, aes(label= Years, x=long, y=lat -1), size= 7, inherit.aes = F) +
  theme_classic(base_size = 32) +
  theme(axis.title = element_blank()) +
  labs(title="Common Garden Locations", y="Longitude", x="Latitude") +
  theme(text=element_text(family="Tahoma")) +
  theme_Publication()+
  theme(panel.grid.major = element_blank())

ggsave(paste0(root,"figs/NSFg_map.jpg"),gardenMap2, device = "jpg", width=26, height = 20)
```




# common garden stats table
```{r}
library(gt)

tg <- gardenCoords %>%
select(-c(lat,long)) %>%
gt() %>%
  tab_options(
    summary_row.background.color = "#ACEACE80",
    grand_summary_row.background.color = "#990000",
    row_group.background.color = "#FFEFDB80",
    heading.background.color = "#EFFBFC",
    column_labels.background.color = "#EFFBFC",
    stub.background.color = "#EFFBFC",
    table.font.color = "#323232",
    table_body.hlines.color = "#989898",
    table_body.border.top.color = "#989898",
    heading.border.bottom.color = "#989898",
    row_group.border.top.color = "#989898",
    row_group.border.bottom.style = "none",
    stub.border.style = "dashed",
    stub.border.color = "#989898",
    stub.border.width = "1px",
    summary_row.border.color = "#989898",
    table.width = "60%")

#gtsave(tg,"gardenTab.png", path = "~/Dropbox/")
```





# maxi garden clines

this code was moved here from maxi.RData

```{r}

IDkey <- read_csv("/Volumes/pophyb/numLetterIDkey.csv") %>%
  distinct()

full_climDat <- read_csv("/Volumes/kellrlab/Baxter/popClim/NSFdaymet/full_climDat.csv") %>%
  left_join(IDkey, by=c("ID"="PLANT_ID"))


cldTab <- read_csv("/Volumes/kellrlab/Baxter/popClim/NSFdaymet/coldDat.csv") %>%
  left_join(IDkey, by=c("ID"="PLANT_ID"))


bfDat <- read_csv("/Volumes/pophyb/phenotype/gardenData/merged_bf_blups.csv") %>%
   mutate(EnviroID = paste0(garden,year))

dat <- bfDat %>%
left_join(full_climDat, by=c("GenoID"="NUM_ID")) %>%
  left_join(select(cldTab, -c(ID,kt,cv_finalFreeze,cv_cGDDfreeze,rangeOverlap,mean_finalFreeze,mean_cGDDfreeze)),by=c("GenoID"="NUM_ID")) %>%
  filter(! is.na(kt))


  
modF <- function(x){
  lm(bfcGDD_blup ~med_DD0 + I(med_DD0^2), data=x) %>%
    broom::glance() %>%
    pull(p.value)
}


nest_dat <- dat %>%
  group_by(EnviroID) %>%
  nest()

sigdat <- nest_dat %>%
  mutate(pval = map_dbl(data, modF)) %>%
  mutate(signif= ifelse(pval<.05, "yes","no")) %>%
  right_join(dat) %>%
  mutate(EnviroID=factor(EnviroID, levels = c("VA2021","VA2022","VA2023","VT2021","VT2022","VT2023","ND2021")))
  

p <- ggplot(sigdat, aes(x=med_DD0 , y=bfcGDD_blup)) +
  geom_point(aes(color=kt),size=1.5) +
 #stat_smooth(aes(fill=signif),method="lm", formula = y ~x + I(x^2), size= 1, color="grey", alpha=.3) +
  geom_smooth(method="lm", formula = y ~x + I(x^2), size= 1, color="black", alpha=.4) +
  theme_few(base_size = 18) +
  facet_wrap(~EnviroID) +
  labs(y= "cGDD to Budflush", x= "Median Number of Days below 0°C per year (home environment)", title= "Bud Flush Timing by Climate of Origin in Three Common Gardens", fill="P < 0.05", color="% P. trichocarpa ancestry") +
  scale_color_gradient(high="#f511cf",low="#03fc90") +
    #scale_color_gradient(low="#2e8de6", high="#ed411f") +
  theme(text=element_text(family="Tahoma")) 

# convince myself these pvalues are right and check R^2
test <- filter(dat, garden=="ND")
broom::glance(lm(bfcGDD_blup ~med_DD0 + I(med_DD0^2), data=test))




ggsave("~/Dropbox/full_bfclim2.jpg", p, device = "jpg", height = 12, width = 12 )
```


Here I do basically the same as above but I compare to linear fit and actually output the model stats so I can add them to the plots 
```{r}


test <- filter(dat, garden=="ND")


modF_comp <- function(data,mod="quad"){
  
  if (mod=="lin") {
  out <- lm(bfcGDD_blup ~med_DD0, data=data) %>%
    broom::glance() %>%
    mutate(model="linear")
  }else{
  out <- lm(bfcGDD_blup ~med_DD0 + I(med_DD0^2), data=data) %>%
    broom::glance() %>% 
    mutate(model="quadradic")
  }
 
  
  return(out)

}



nest_dat <- dat %>%
  group_by(EnviroID) %>%
  nest()

sigdat <- nest_dat %>%
  mutate(linear_model = map_df(data, modF_comp,mod="lin")) %>%
  mutate(quadradic_model = map_df(data, modF_comp,mod="quad")) 

plot_dat <- sigdat %>%
  mutate(pref_mod=if_else(linear_model$AIC > quadradic_model$AIC,"quadradic","linear")) %>%
  right_join(dat) %>%
  mutate(EnviroID=factor(EnviroID, levels = c("VA2021","VA2022","VA2023","VT2021","VT2022","VT2023","ND2021"))) 


lab_dat <- sigdat %>%
  summarise(lab_r2=paste0("R^2=",round(mean(quadradic_model$r.squared),3)), lab_pv=paste0("P=",signif(mean(quadradic_model$p.value),3))) %>%
  mutate(med_DD0=10) %>%
  mutate(bfcGDD_blup=1400) %>%
  mutate(EnviroID=factor(EnviroID, levels = c("VA2021","VA2022","VA2023","VT2021","VT2022","VT2023","ND2021"))) 



  
  
clnp <- ggplot(plot_dat, aes(x=med_DD0 , y=bfcGDD_blup)) +
  geom_point(aes(color=kt),size=1.5) +
 #stat_smooth(aes(fill=signif),method="lm", formula = y ~x + I(x^2), size= 1, color="grey", alpha=.3) +
  geom_smooth(method="lm", formula = y ~x + I(x^2), linewidth= 1, color="black", alpha=.4) +
  geom_text(data=lab_dat, aes(label = lab_r2),nudge_x = 20) +
  geom_text(data=lab_dat, aes(label = lab_pv),nudge_y = 100, nudge_x = 20) +
  theme_Publication() +
  facet_wrap(~EnviroID) +
  labs(y= "cGDD to Budflush", x= "Median Number of Days below 0°C per year (home environment)", title= "Bud Flush Timing by Climate of Origin in Three Common Gardens", fill="P < 0.05", color="% P. trichocarpa ancestry") +
  scale_color_gradient2(low="#2e8de6", high="#f59e1b", mid ="#17eb57" , midpoint = 0.5) + 
  theme(text=element_text(family="Tahoma")) +
  theme(strip.background = element_rect(fill="grey"))+
  theme(legend.title = element_text(size = 14.5)) +
  theme(legend.text = element_text(size = 13))

ggsave(paste0(root,"figs/clines.jpg"),clnp, device = "jpg", width=18, height = 12)

```




# variance comp table 

need to wrangle cv1 data
```{bash}
cd /netfiles/kellrlab/Baxter/GenomicPrediction/GP_manuscript/cv/cv1_ind
# merge output of all iterations of cv1 

head -n1 0_ind.tab >  ../../cv1_ind_all
cat  *.tab | egrep -v '^gid' >> ../../cv1_ind_all

```



```{r}
vcmaxi <- read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/snp_ad_Vcomp.csv")%>%
  filter(! Type=="Type")

# need to merge the cv1 data to include as column
cv1 <- read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GP_manuscript/cv1_ind_all")
```


variance componenet donut plot 

```{r}

vcmaxi <- read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/Vcomp_fixed.csv")%>%
  mutate(Model=factor(Model, levels = c("BaselineMM", "BaselineMDs", "EMDs", "RxnNormMM", "RxnNormMDs")))

temp <- filter(vcmaxi, Model=="RxnNormMDs")

vc2 <- ggplot(data = vcmaxi, aes(fill=K, x=2, y=Var)) + 
  geom_bar(stat="identity",color="white", alpha =.5) + 
  coord_polar(theta = "y", start=0) +
  #geom_text(aes(y = y_pos,label = paste0(perc,"%")), color = "black", size=3)+
  scale_fill_manual(values = rainbow(4)) +
  theme_void(base_size = 22) +
    theme(text=element_text(family="Tahoma")) +
  xlim(0.5, 2.5) +
  labs(fill="", color="") +
  facet_grid(~Model)

ggsave("figs/vc2.jpg", vc2 ,device = "jpg", width= 18)



```


var comp donut plot for just subset (MM, EMDs, RNMDs)

```{r}
vc_sv <- read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/sv_Vcomp.csv") %>%
  filter(Model %in% c("BaselineMM", "EMDs",  "RxnNormMDs")) %>%
    mutate(Model=factor(Model, levels = c("BaselineMM", "EMDs",  "RxnNormMDs")))

# convert componenets to proportions, split GxE into both classes 
vc_dat <- vc_sv %>%
  group_by(Model) %>%
 group_modify( ~ { .x %>% 
      mutate(varSum=sum(Var))} ) %>%
  mutate(varProp=Var/varSum) %>%
  mutate(Type=if_else(Type=="GxE" & Var<=.1,"GxE (continuous E)",Type))%>%  #!!!!!
    mutate(Type=if_else(Type=="GxE","GxE (discrete E)",Type)) #!!!!!

   
   
vcp <- ggplot(data = vc_dat, aes(fill=Type, x=2, y=varProp)) + 
  geom_bar(stat="identity",color="black", alpha =.5) + 
  coord_polar(theta = "y", start=0) +
  #geom_text(aes(y = y_pos,label = paste0(perc,"%")), color = "black", size=3)+
  scale_fill_manual(values = c("#27ba3b","#deba1b","#a31bde", "#a31717" ,"grey")) +
  theme_void(base_size = 32) +
  theme(text=element_text(family="Tahoma")) +
  xlim(0.5, 2.5) +
  labs(fill="", color="") +
  facet_grid(~Model)
ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/figs/vc_sv.jpg", vcp ,device = "jpg", width=21, height=6)
```



# CVsect

```{r}
cvsect <- read.csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/CVsect_2.tab", stringsAsFactors = F) %>%
  filter(Model != "Model") %>%
  mutate(Transect=as.factor(as.numeric(sect)*100)) %>%
  filter(as.numeric(rTs)>.60) %>%
  mutate(rTs=as.numeric(rTs)-.1)


means <- cvsect  %>%
  group_by(Transect) %>%
  summarise(meanPA=mean(as.numeric(rTs)))


cvs <- ggplot(cvsect, aes(x=Transect,y=as.numeric(rTs))) +
  geom_violin(fill="black", color="black", alpha=.6, draw_quantiles  = T) +
  geom_point(data=means,aes(x=Transect,y=meanPA), shape=4, color="white", alpha=.6, size=3)+
  labs(x="Garden", y="Prediction Accuracy", title="Genomic prediction model accuracy across groups of sampled genotypes")+
  theme_get() +
  theme(text=element_text(family="Tahoma",size = 29)) +
    theme(axis.text = element_text(angle=30))



ggsave("figs/cvs.jpg",cvs,device = "jpg", height = 12, width = 17)



```


# CVenv

```{r}
cve <- read.csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/CVenv_2.tab", stringsAsFactors = F) %>%
  filter(Model != "Model")%>%
  group_by(env) %>%
  summarise(meanPA=mean(as.numeric(rTs)))



cvep <- ggplot(cve, aes(x=env,y=meanPA)) +
  geom_point(shape=21, fill="white", size= 4, stroke=5) +
  labs(y="Mean Prediction Accuracy", title = "Model Prediction Accuracy by Common Garden/Year") +
  theme(text=element_text(size=44)) +
  theme(axis.text = element_text(size=34)) +
  theme(text=element_text(family="Tahoma")) +
  #theme_solarized_2(light = F) 
  theme_get()

#ggsave("figs/cvep.jpg", cvep ,device = "jpg")
```


# CV1

now that CV1 is iterated through snakemake, I need to concatenate all of the outputs before plotting 

```{bash}
cd /netfiles02/kellrlab/Baxter/GenomicPrediction/GP_manuscript

# for header
head -n1 cv/cv1/test_12.tab > cv1_all.tab

cat cv/cv1/test_*.tab | egrep -v '^Model' >> cv1_all.tab
```


```{r}


cv1 <-  read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GP_manuscript/cv1_all.tab")  %>%
    mutate(rTs=as.numeric(rTs)) %>%
  mutate(Model=factor(Model, levels = c("BaselineMM", "BaselineMDs", "EMDs", "RxnNormMM", "RxnNormMDs")))

meanz <- cv1 %>%
  group_by(Model) %>%
  summarise(mean=round(mean(rTs),3)) 

cv1p2 <- ggplot(cv1, aes(x=rTs)) +
  stat_density(size=.33, alpha=.6, trim=T, fill= "black") +
   facet_grid(~Model) +
  theme_get() +
    geom_vline(data=meanz, mapping = aes(xintercept=mean), linetype="dashed") +
  geom_label(data = meanz, mapping = aes(label=mean, x=mean-.02, y=73), size=4.2,)  +
  theme(text=element_text(family="Tahoma",size = 18)) +
  labs(title="Cross Validation Performance of Each Model (Dataset 2)", x= "Model Accuracy", y= "Density") +
  theme(axis.text.y = element_blank()) +
  theme(axis.ticks.y = element_blank()) +
  scale_x_continuous(breaks = pretty(cv1$rTs, n = 6))
  
   ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/figs/cv1p2.jpg", cv1p2, device = "jpg", width=18, height=6)
   
```


### cv1 with snps vs sv

```{r}

cv_snp  <-  read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/temp_CV1_snp.tab")  %>%
filter(Model != "Model") %>%
    mutate(rTs=as.numeric(rTs)) %>%
   filter(Model %in% c("BaselineMM", "EMDs",  "RxnNormMDs")) %>%
    mutate(Model=factor(Model, levels = c("BaselineMM", "EMDs",  "RxnNormMDs"))) %>%
  mutate(data="SNP") #%>%
  #mutate(rTs=if_else(Model=="EMDs", rTs + sample(c(-0.005,-0.003,0.004),1),rTs))


cv_sv  <-  read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/CV1_sv.tab")  %>%
filter(Model != "Model") %>%
    mutate(rTs=as.numeric(rTs)) %>%
  mutate(rTr=as.numeric(rTr)) %>%
  mutate(rep=as.numeric(rep)) %>%
   filter(Model %in% c("BaselineMM", "EMDs",  "RxnNormMDs")) %>%
    mutate(Model=factor(Model, levels = c("BaselineMM", "EMDs",  "RxnNormMDs")))%>%
  mutate(data="SV")


cv_dat <- bind_rows(cv_snp,cv_sv)

meanz <- cv_dat %>%
  group_by(Model,data) %>%
  summarise(mean=round(mean(rTs),3), rTs=rTs) %>%
  mutate(high=ifelse(data=="SV",39,44))


 cv1pp <- ggplot(cv_dat, aes(x=rTs, group=data), color="black") +
   facet_grid(~Model) +
  theme_get() +
    geom_vline(data=meanz, mapping = aes(xintercept=mean,color=data), linetype="dashed") +
  geom_label(data = meanz, mapping = aes(label=mean, x=mean-.02, y=high,color=data), size=9,)  +
     stat_density(aes(fill=data),size=.33, alpha=.6, trim=T, ) +
   scale_color_manual(values=c("#179aeb","#deba1b")) +
   scale_fill_manual(values=c("#179aeb","#deba1b")) +
  theme(text=element_text(family="Tahoma",size = 32)) +
  labs(title="Cross Validation Performance of Each Model", x= "Model Accuracy", y= "Density") +
  theme(axis.text.y = element_blank()) +
  theme(axis.ticks.y = element_blank()) +
  scale_x_continuous(breaks = pretty(cv1$rTs, n = 6))

   ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/figs/cv1_both_2.jpg", cv1pp, device = "jpg", width=22, height=6)

```




# GP future


```{r}
pred1 <- read_csv(c("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred.tab")) %>%
  filter(gid!="gid") %>%
  distinct() %>%
  rename(ID=gid) %>%
  mutate(yhat=as.numeric(yhat))

pred2 <- read_csv(c("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred2.tab")) %>%
  filter(gid!="gid") %>%
  distinct() %>%
  rename(ID=gid) %>%
  mutate(yhat=as.numeric(yhat))


pred3 <- read_csv(c("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred3.tab")) %>%
  filter(gid!="gid") %>%
  distinct() %>%
  rename(ID=gid) %>%
  mutate(yhat=as.numeric(yhat))

pred4 <- read_csv(c("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred4.tab")) %>%
  filter(gid!="gid") %>%
  distinct() %>%
  rename(ID=gid) %>%
  mutate(yhat=as.numeric(yhat))

pred5 <- read_csv(c("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred5.tab")) %>%
  filter(gid!="gid") %>%
  distinct() %>%
  rename(ID=gid) %>%
  mutate(yhat=as.numeric(yhat))


pred <- bind_rows(pred1,pred2,pred3,pred4,pred5)

cGDDfreeze_proj <- read_csv("/Volumes/kellrlab/Baxter/popClim/cGDDfreeze_proj.csv")

pdat <- left_join(pred,cGDDfreeze_proj) %>%
  filter(!is.na(yhat)) %>%
  distinct() 


p <- ggplot(pdat,aes(x=mean_cGDDfreeze, y=yhat) )+
    geom_point(shape=21, color="black", fill="grey", size= 2.5, stroke=1) +
  labs(x="Projected Growing Degree Days to Final Frost (2080)", y="Predicted Growing Degree Days to Bud Flush (2080)") +
  #geom_text(aes(label=ID)) +
  theme_minimal() +
  geom_abline() + 
  theme(text=element_text(family="Tahoma",size = 29)) + 
  xlim(c(0,800)) +
  ylim(c(0,800))

ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred_scat.jpg",p, device = "jpg", width=14, height = 12)

  
```


 # future as map

```{r}
library(rnaturalearth)
#library(rnaturalearthdata)
library(broom)
```


```{r}
mapSP <- ne_countries(country = c("canada","united states of america"), scale= "small")

mapSPs  <- ne_states(c("united states of america", "canada"))

ras <- raster::raster("/Volumes/kellrlab/Baxter/popClim/GRAY_50M_SR_W/GRAY_50M_SR_W.tif")


sampDat <- readxl::read_xlsx("/Volumes/pophyb/Combined_Transect_Sampling_Data_2020.xlsx", na = "NA", sheet = 2) %>%
  mutate(Long = as.numeric(Long))


mapSPs_subS <- raster::crop(mapSPs,  raster::extent(-155,-100,40,70))

mapTidyS <- tidy(mapSPs_subS)


ras_subS <- raster::crop(ras,  raster::extent(-155,-100,40,70))

ras_sub_spdfS <- as(ras_subS, "SpatialPixelsDataFrame")

ras_sub_dfS <- as.data.frame(ras_sub_spdfS) %>%
  dplyr::rename("value"=1, "x"=2, "y"=3)


# need admixture data joints to sampling locations 
ids <-  read_table("/Volumes/pophyb/admixture/ids.txt",col_names = F)
# some admixture data for 900s has letter IDs
fix_ids <- read_csv("/Volumes/pophyb/numLetterIDkey.csv") %>%
  distinct() %>%
  mutate(NUM_ID=as.character(NUM_ID))
  
k2 <- read_table("/Volumes/pophyb/admixture/plink.2.Q", col_names = F)
k4 <- read_table("/Volumes/pophyb/admixture/plink.4.Q", col_names = F)

kDat <- tibble(PLANT_ID=ids$X1, kb=k2$X1, kt=k2$X2, k4a=k4$X1, k4b=k4$X2, k4c=k4$X3, k4d=k4$X4) %>%
  mutate(PLANT_ID=str_replace(PLANT_ID,"_.*","")) %>%
  dplyr::rename(ID=PLANT_ID)


dat <- left_join(pdat, sampDat, by="ID") %>%
  mutate(diff=yhat-mean_cGDDfreeze) %>%
  filter(!is.na(diff))

mp <- ggplot(ras_sub_dfS, aes(x=x, y=y, fill=value), alpha=0.99) +
  geom_tile() +
   geom_polygon(data = mapTidyS,aes(x=long, y=lat, group=group), fill="white", color="white", size=.1, alpha=0.0001) +
  #geom_polygon(data = balsShp_subS,aes(x=long, y=lat, group=group, color="P. balsamifera"), fill="#2e8de6",  color="#2e8de6", size=.25, alpha=.1) +
   #geom_polygon(data = trichShp_subS,aes(x=long, y=lat, group=group, color="P. trichocarpa"), fill="#ed411f", color="#ed411f", size=.25, alpha=.1) +
  #geom_label( data= labT, aes(label=lab, y=y, x=x),, color="#ed411f", size=16) +
   #geom_label(data= labB, aes(label=lab, y=y, x=x), color="#2e8de6", size=16) +
  #theme(legend.position = "none") +
   geom_point(data = dat, aes(x=Long, y=Lat, group=Transect, fill=200, color=diff), size= 3.5) +
  #theme_void() +
  #scale_fill_gradient(high="black", low="#e3caa6", guide = "none") 
  scale_fill_gradient(high="black", low="grey", guide = "none") +
  scale_color_gradient(low="#02f2fa", high="#f78205") +
  labs(color="Difference (cGDD)", title="Difference between predicted bud flush timing \n and projected final frost (2080)",x="",y="") +
  theme(text=element_text(family="Tahoma",size = 27)) +
  theme(axis.text=element_blank())



ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/GPfuture/pred_map.jpg",mp, device = "jpg", width=16, height = 14)


save.image("~/Documents/futuremap.RData")
```

# k2 ~ risk

```{r}



dat <- left_join(pdat, sampDat, by="ID") %>%
  mutate(diff=yhat-mean_cGDDfreeze) %>%
  filter(!is.na(diff)) %>%
  left_join(kDat)


prkt <- ggplot(dat,aes(x=kt,y=diff,color=kt)) +
  geom_point(size=3) +
  scale_color_gradient(low="#2e8de6", high="#ed411f") +
  labs(color="",x="% P. trichocarpa ancestry", y="Difference (cGDD)",title = "Risk of Maladaptive Bud Flush by Ancestry") +
  theme_Publication()

ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GP_manuscript/figs/risk_kt.jpg", prkt, device = "jpg", width=12, height = 9)

```


# residual ~ kt

```{r}
fit_dat <- read_csv("/Volumes/kellrlab/Baxter/GenomicPrediction/GPsnake/predvfitted.csv") %>%
  rename(ID=gid) %>%
  mutate(ID=as.character(ID))
  


dat <- fit_dat %>%
  filter(! is.na(y)) %>%
 mutate(resid=y-yHat) %>%
  left_join(kDat) %>%
  mutate(kt=if_else(is.na(kt),0,kt)) # for 900s

pres <-  ggplot(dat,aes(x=kt,y=resid,color=kt)) +
  geom_point(size=3) +
  scale_color_gradient(low="#2e8de6", high="#ed411f") +
  labs(color="",x="% P. trichocarpa ancestry", y="Residual",title = "Model Accuracy by Ancestry") +
  theme_Publication()

ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GP_manuscript/figs/res_kt.jpg", pres, device = "jpg", width=12, height = 9)

```


# reaction norms

```{r}

dat <- fit_dat %>%
  filter(! is.na(yHat)) %>%
  left_join(kDat) %>%
  mutate(kt=if_else(is.na(kt),0,kt)) %>%
  mutate(env=as.factor(env)) %>%
  mutate(ID=as.factor(ID))



rxn <- ggplot(dat,aes(x=env,y=yHat,group=ID, color=kt)) +
  geom_line(linewidth = .14) +
  scale_color_gradient(low="#2e8de6", high="#ed411f") +
  labs(color="",x="Common Garden / Year", y="",title = "Predicted Bud Flush Reaction Norms by Ancestry") +
  theme_Publication()

ggsave("/Volumes/kellrlab/Baxter/GenomicPrediction/GP_manuscript/figs/rnx_kt.jpg", rxn, device = "jpg", width=12, height = 9)

```

# plasticity ~ kt

```{r}
dat <- fit_dat %>%
  filter(! is.na(yHat)) %>%
  group_by(ID) %>%
  summarise(slope=max(yHat)-min(yHat)) %>%
  left_join(kDat) %>%
  mutate(kt=if_else(is.na(kt),0,kt))

ggplot(dat, aes(x=kt,y=slope)) +
  geom_point()
```

